extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java

```diff
@@ -522,7 +522,12 @@ private void addSpringValueAnnotations(AnnotationTarget target, AnnotationInstan
             return;
         }
         String defaultValue = null;
-        String propertyName = annotationValue.asString().replace("${", "").replace("}", "");
+        String stringValue = annotationValue.asString();
+        if (stringValue.contains("#{")) {
+            throw new IllegalArgumentException(
+                    "SpEL expressions are not supported when using " + SPRING_VALUE_ANNOTATION.toString());
```

Comment:
Could you print the expression in the error message?

Reply:
Good idea. Done